src/error.rs

```diff
@@ -4,20 +4,43 @@
 //! use, these errors should be as tiny
 //! and precise as possible.
 
+#[cfg(feature = "std")]
+use core::fmt::Display;
+
 /// Error that indicates an invalid event id was detected.
 ///
 ///
 /// event ids must be greater than 0 and less than EventId::MAX_USER_ID
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub struct InvalidEventId;
 
+#[cfg(feature = "std")]
+impl std::error::Error for InvalidEventId {}
+
+#[cfg(feature = "std")]
+impl Display for InvalidEventId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("Invalid Event Id")
+    }
+}
+
 /// Error that indicates an invalid probe id was detected.
 ///
 ///
 /// probe ids must be greater than 0 and less than ProbeId::MAX_ID
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub struct InvalidProbeId;
 
+#[cfg(feature = "std")]
+impl std::error::Error for InvalidProbeId {}
+
+#[cfg(feature = "std")]
```

Comment:
The `Display` trait is available in `core`, should be able to implement without feature-flagging.